### DISCUSSION TITLE
[ComVisible(true)] in Ribbon class

### DISCUSSION THREAD

**Discussion by schebotar** [Excel-DNA/ExcelDna#620] [Q&A] [answered]
I'm curious, why do we need [ComVisible(true)] attribute in Ribbon class?
Just tried to run simple panel without it and it works fine.

**Comment by govert** [comment, marked as answer]
In the past, the default C# project template put an attribute in the AssemblyInfo.cs that sets
```
[assembly:ComVisible(false)]
```

With this common assembly-wide attribute in place, you need to explicitly set your class to be `ComVisible(true)`. Without the assembly attribute, the default is for any public class to be `ComVisible`, and so the extra attribute does nothing.
